### PR TITLE
Fix replays not failing when last tick is the cause

### PR DESCRIPTION
**Pull request by ZehMatt**
With the current logic there is an edge case if the last tick has the faulty checksum. After the last tick the _mode is no longer set to Playing and IsPlaybackStateMismatching would always result false.

I changed it to wait for the playback to complete and then test if the state is mismatching. This might cause the CI to fail now with out of date replays.

Later me: Yes the CI did fail as expected, also updating the replay version in this PR to have them all working again.

### CODE REVIEW

File: test/tests/ReplayTests.cpp [hunk 1]
```diff
@@ -98,8 +98,9 @@ TEST_P(ReplayTests, RunReplay)
     while (replayManager->IsReplaying())
     {
         gs->UpdateLogic();
-        ASSERT_TRUE(replayManager->IsPlaybackStateMismatching() == false);
     }
+    ASSERT_FALSE(replayManager->IsReplaying());
+    ASSERT_FALSE(replayManager->IsPlaybackStateMismatching());
```
Comment (duncanspumpkin) [inline test/tests/ReplayTests.cpp:103]: Is there any point continuing the replay if there has been a mismatch

Comment (ZehMatt) [reply test/tests/ReplayTests.cpp:103]: I thought about that as well, I can stop the playback in CheckState I guess.

Comment (duncanspumpkin) [reply test/tests/ReplayTests.cpp:103]: Also is it possible for the checksum to be wrong at the start but then become correct in the subsequent tick? Not sure if that is possible.

Comment (ZehMatt) [reply test/tests/ReplayTests.cpp:103]: I don't think that's possible, they are checked after the update.

**Pull request merged**